src/report/example.rs

```diff
@@ -0,0 +1,82 @@
+use std::convert::From;
+
+#[cfg(feature = "expectest_compat")]
+use expectest::core::TestResult as ExpectestResult;
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Failure {
+    pub message: Option<String>,
+}
+
+impl Failure {
+    pub fn new<S>(message: Option<S>) -> Self
+    where
+        String: From<S>,
+    {
+        Failure { message: message.map(|s| s.into()) }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum ExampleReport {
+    Success,
+    Ignored,
+    Failure(Failure),
+}
+
+impl ExampleReport {
+    pub fn err<S>(message: Option<S>) -> Self
+    where
+        String: From<S>,
+    {
+        ExampleReport::Failure(Failure::new(message))
+    }
+
+    pub fn is_ok(&self) -> bool {
+        *self == ExampleReport::Success
+    }
+
+    pub fn is_err(&self) -> bool {
+        *self != ExampleReport::Success
+    }
+}
+
+impl From<()> for ExampleReport {
+    fn from(_other: ()) -> ExampleReport {
+        ExampleReport::Success
+    }
+}
+
+impl From<bool> for ExampleReport {
+    fn from(other: bool) -> ExampleReport {
+        if other {
+            ExampleReport::Success
+        } else {
+            ExampleReport::Failure(Failure::new(Some("assertion failed: `expected true`")))
```

Comment:
What else would you put? The `assertion failed: …` is meant to match the format of both rust's native test matchers (`assert!()`), as well as expectest's (as of [this](https://github.com/zummenix/expectest/pull/34) PR).